tests/integration_test/run_integration_test.py

```diff
@@ -0,0 +1,142 @@
+# Util imports
+from input_output.file_output import load_graph_list_from_filepath, create_csv_file, write_csv_line
+from input_output.sys_output import fail, passed
+import tkinter as tk
+from tkinter import filedialog
+from time import time
+# Algorithm imports
+from tests.integration_test.settings import *
+from tests.integration_test.solutions import *
+from tests.integration_test.isomorphism_problem import are_isomorph, amount_of_isomorphisms
+
+"""
+General integration test for the Graph Isomorphisms problem.
+Settings of the algorithm must be changed in > settings.py <
+"""
+
+"""
+FILE INPUT
+"""
+# Select files to evaluate
+root = tk.Tk()
+root.withdraw()
+file_paths = filedialog.askopenfilenames()
+
+"""
+FILE OUTPUT
+"""
+# If csv must be created, create an empty file here
+if write_to_csv:
+    csv_filepath = create_csv_file("isomorphism-problem-test")
+    # Write first row with column names
+    csv_column_names = ['file', 'graph1', 'graph2',
+                        'are_isomorph expected result', 'are_isomorph actual result', 'passed', 'are_isomorph processing time (s)',
+                        'amount_of_isomorphisms expected result', 'amount_of_isomorphisms actual result', 'passed', 'amount_of_isomorphisms processing time (s)']
+    write_csv_line(csv_filepath, csv_column_names)
+
+"""
+ACTUAL TEST
+"""
+# For every file that needs to be evaluated, all combinations of graphs within that file are evaluated.
+graph_count = 0
+error_count = 0
+for filepath in file_paths:
+    graphs = load_graph_list_from_filepath(filepath)
+    filename = (filepath.split("/")[-1]).split(".")[0]
+    solution_map = {}
+    if filename in solution:
+        solution_map = solution[filename]
+
+    for i in range(0, len(graphs) - 1):
+        for j in range(i + 1, len(graphs)):
+            G = graphs[i]
+            H = graphs[j]
+
+            if console_pass or console_fail:
+                print('---------------------------')
+                print("Statistics of " + filename + "-" + str(i) + "_" + str(j) + ":")
+                print('---------------------------')
+
+            start_isomorph = time()
+            are_isomorph_actual = are_isomorph(G.copy(), H.copy())
```

Comment:
Hier is het niet helemaal eerlijk dat de tijd voor het kopiëren bij de tijd voor are_isomorph in zit. Mooier is denk ik om ze daarboven te kopiëren en die hier te gebruiken en dan de originele gewoon zoals hieronder al gedaan wordt.

Reply:
Check